<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ appliedepi</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">appliedepi</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/applied_epi_usage.html">applied_epi_usage</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jagadeeshpuvvula/appliedepi/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="annotate_and_save.html">annotate_and_save()</a></code> </p>
        </td>
        <td><p>CpG annotate and save csv files</p></td>
      </tr><tr><td>
          <p><code><a href="assignGroup.html">assignGroup()</a></code> </p>
        </td>
        <td><p>Add KEGG superpathways to the pathways enriched from the PEA analysis
Run pathway_list before using this function or will result in errors</p></td>
      </tr><tr><td>
          <p><code><a href="bind_mwas_res_from_folder.html">bind_mwas_res_from_folder()</a></code> </p>
        </td>
        <td><p>This function will bind rows across the csv files in a given target folder</p></td>
      </tr><tr><td>
          <p><code><a href="bkmr_cohort.html">bkmr_cohort()</a></code> </p>
        </td>
        <td><p>BKMR wrapper performing cohort specific loops. Load Matrix and bindrcpp packages</p></td>
      </tr><tr><td>
          <p><code><a href="bkmr_sex.html">bkmr_sex()</a></code> </p>
        </td>
        <td><p>BKMR wrapper performing sex specific loops. Load Matrix and bindrcpp packages</p></td>
      </tr><tr><td>
          <p><code><a href="calculatePvalCutoff.html">calculatePvalCutoff()</a></code> </p>
        </td>
        <td><p>Manually get p-value cutoffs based on the 500 features with lowest p-values
use along with the runMummichog function</p></td>
      </tr><tr><td>
          <p><code><a href="combine_csv_files.html">combine_csv_files()</a></code> </p>
        </td>
        <td><p>Combines mummuchog output files into a single dataframe</p></td>
      </tr><tr><td>
          <p><code><a href="enrichment_summary.html">enrichment_summary()</a></code> </p>
        </td>
        <td><p>ewas enrichemnt summary</p></td>
      </tr><tr><td>
          <p><code><a href="ewas_loop_pll.html">ewas_loop_pll()</a></code> </p>
        </td>
        <td><p>Performs EWAS using CpGassoc library. Allows flexibility to loop for multiple exposure variables and methylation data from multiple tissues.</p></td>
      </tr><tr><td>
          <p><code><a href="ewas_summary.html">ewas_summary()</a></code> </p>
        </td>
        <td><p>Summarize results from EWAS analysis</p></td>
      </tr><tr><td>
          <p><code><a href="extract_data_from_folder.html">extract_data_from_folder()</a></code> </p>
        </td>
        <td><p>Extract results from the bkmr RDA files</p></td>
      </tr><tr><td>
          <p><code><a href="extract_pips.html">extract_pips()</a></code> </p>
        </td>
        <td><p>Extract bkmr PIPs from all RDA files - prep for Shiny object</p></td>
      </tr><tr><td>
          <p><code><a href="extract_weights.html">extract_weights()</a></code> </p>
        </td>
        <td><p>Extract chemical weights contributed to the joint association using non bootstrap quantile g-computation.</p></td>
      </tr><tr><td>
          <p><code><a href="get_gcomp_estimates.html">get_gcomp_estimates()</a></code> </p>
        </td>
        <td><p>Extract results from the qgcomp output RDA files. Iterates though the RDA files and compiles results in tibble format</p></td>
      </tr><tr><td>
          <p><code><a href="lm_func.html">lm_func()</a></code> </p>
        </td>
        <td><p>Multiple linear regression wrapper function that will take a list of exposure and outcome variables. Returns a matrix of results between each combination</p></td>
      </tr><tr><td>
          <p><code><a href="match_sequence.html">match_sequence()</a></code> </p>
        </td>
        <td><p>match sequence - validation step before getting into MWAS loop
match the exposure dataset with the metabolome sequence
metabolome each observation is a feature and each variable is a subject
Whereas the exposure/covariate dataset each observation is a subject</p></td>
      </tr><tr><td>
          <p><code><a href="missing_data_summary.html">missing_data_summary()</a></code> </p>
        </td>
        <td><p>Missing data summary, prints frequency and count</p></td>
      </tr><tr><td>
          <p><code><a href="mwas_lm_loop.html">mwas_lm_loop()</a></code> </p>
        </td>
        <td><p>mwas loop - perform linear regression between each feature and exposure metric provided in the input
metabolome data will be transposed in the function
input metabolome features and exposures are in different dataframes
runtime 17,000 * 70 * 17 matrix takes around 10 mins</p></td>
      </tr><tr><td>
          <p><code><a href="perform_gene_enrichment.html">perform_gene_enrichment()</a></code> </p>
        </td>
        <td><p>Gene enrichment for EWAS files</p></td>
      </tr><tr><td>
          <p><code><a href="print_version_info.html">print_version_info()</a></code> </p>
        </td>
        <td><p>Prints detailed summary of the R environment including dependencies</p></td>
      </tr><tr><td>
          <p><code><a href="process_methylation_data.html">process_methylation_data()</a></code> </p>
        </td>
        <td><p>Process methylation data using minfi pipeline. This function will drop samples with GT 5 percent high detection p-values, probes with at least one high detection p-value, Sex chromosomes, loci associated with single-nucleotide polymorphisms and cross-hybridizing probes. Additionally, will preprocess using Noob for background correction and dye bias normalization, BMIQ for probe design bias, batch correction using ComBat.</p></td>
      </tr><tr><td>
          <p><code><a href="qgcomp_func.html">qgcomp_func()</a></code> </p>
        </td>
        <td><p>Quantile g-computation wrapper function to automate analysis for multiple outcomes. Additionally, allows flexibility to perform stratified analysis.</p></td>
      </tr><tr><td>
          <p><code><a href="runMummichog.html">runMummichog()</a></code> </p>
        </td>
        <td><p>Performs Pathway Enrichment Analysis (PEA) using mummichog function available from metaboAnalystR library
This function also prints the p-value theshold on scree that is being used for the analysis</p></td>
      </tr><tr><td>
          <p><code><a href="split_and_export.html">split_and_export()</a></code> </p>
        </td>
        <td><p>Split MWAS LM result files from a single dataframe to .txt files for PEA mummichog</p></td>
      </tr><tr><td>
          <p><code><a href="summarize_variables.html">summarize_variables()</a></code> </p>
        </td>
        <td><p>Summarize categorical and continuous variables</p></td>
      </tr><tr><td>
          <p><code><a href="var_selec.html">var_selec()</a></code> </p>
        </td>
        <td><p>Variable selection using group LASSO (gglasso, grpreg that use distinct mathematical approaches), sparse group LASSO, glmnet, stability selection with error control</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jagadeesh Puvvula.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

